src/Plaid/PlaidClient.cs

```diff
@@ -165,6 +167,37 @@ public sealed partial class PlaidClient
 			}
 		}
 
+		public readonly async Task<FileResponse> ParseFileResponseAsync()
+		{
+			// NOTE: We do not dispose the HttpResponseMessage here. We pass this duty onto the FileResponse
+			var response = await Message.ConfigureAwait(false);
+			Logger.LogInformation("Completed file request. Url: {Url}, Status Code: {StatusCode}.", Url, response.StatusCode);
+
+			var status = response.StatusCode;
+			if (!response.IsSuccessStatusCode)
+			{
+				var json = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+				var error = ParseError((int)response.StatusCode, json);
+				var badresult = new FileResponse(status, error);
+				response.Dispose();
+
+				return badresult;
+			}
+
+			IEnumerable<KeyValuePair<string, IEnumerable<string>>> inheaders = response.Headers;
+			if (response.Content?.Headers is not null)
+				inheaders = inheaders.Concat(response.Content.Headers!);
+			var headers = inheaders.ToDictionary(x => x.Key, x => x.Value, StringComparer.Ordinal);
+
+			var stream = response.Content == null ? System.IO.Stream.Null : await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+			var result = new FileResponse(status, headers, stream, response!);
+
+			Logger.LogTrace("Completed file request details. Url: {Url}; Response: {@Result}",
```

Comment:
One thing to confirm: whether the object expansion of the `@Result` for logging will read the entire stream and dump it into the log?

Reply:
I confirmed this. It only expands one level deep.

```
{[@Result, {FileResponse { RawJson = , Error = , RequestId = , StatusCode = OK, IsSuccessStatusCode = True, Headers = System.Collections.Generic.Dictionary`2[System.String,System.Collections.Generic.IEnumerable`1[System.String]], Stream = System.IO.MemoryStream }}]}
```

Also we know it didn't read the stream, because if it DID, we'd have gotten errors later trying to read it again.

Reply:
Ok, two final comments. 

1. Let's do this instead here. I think RequestId and StatusCode are going to be the useful data out of this.

```csharp
if (Logger.IsEnabled(LogLevel.Trace))
{
  Logger.LogTrace("...",
    url,
    new { result. RequestId, result.StatusCode, result.IsSuccessStatusCode });
}
```

2. Add a log message in the error branch as well, similar to above, but include the Error in the anonymous object as well.